src/elfpy/markets.py

```diff
@@ -476,16 +466,16 @@ def _remove_liquidity(
         )
         return market_deltas, agent_deltas
 
-    def log_market_step_string(self, pricing_model: PricingModel) -> None:
+    def log_market_step_string(self) -> None:
         """Logs the current market step"""
         # TODO: This is a HACK to prevent test_sim from failing on market shutdown
         # when the market closes, the share_reserves are 0 (or negative & close to 0) and several logging steps break
         if self.market_state.share_reserves <= 0:
             spot_price = str(np.nan)
             rate = str(np.nan)
         else:
-            spot_price = self.get_spot_price(pricing_model)
-            rate = self.get_rate(pricing_model)
+            spot_price = self.get_spot_price()
```

Comment:
It might make sense to make spot price & rate properties now that they dont have arguments